### DISCUSSION TITLE
Elfin-EE11 connection problem

### DISCUSSION THREAD

**Discussion by szenna** [wlcrs/huawei_solar#614] [Connecting to the inverter]
Hi, I would like to ask for your help in setting up Elfin-EE11.
I have set up the inverter and Elfin-EE11 as described. Only on the inverter I failed to set the "RS485 Bus Frame Capture" parameter. I attach all screenshots. When I want to connect in HA, it fails with "No response" error.
On the Elfin-EE11 status page it shows that there is serial port communication, but TCP is only received, not sent. Any ideas? Thanks in advance!
![1](https://github.com/wlcrs/huawei_solar/assets/77939584/35ec8ff9-1724-4b71-869c-79fb83fe7e4f)
![2](https://github.com/wlcrs/huawei_solar/assets/77939584/26fb7438-b18e-49f8-b746-03c6cff3079e)
![3](https://github.com/wlcrs/huawei_solar/assets/77939584/1bfdb52a-5bbb-47de-adf4-b97563ae5924)
![4](https://github.com/wlcrs/huawei_solar/assets/77939584/ee056017-cb0d-44b9-85f1-c0a342e21ced)
<img width="1080" alt="5" src="https://github.com/wlcrs/huawei_solar/assets/77939584/6a338512-7c91-4e14-812d-082e00c71f44">
<img width="1079" alt="6" src="https://github.com/wlcrs/huawei_solar/assets/77939584/ceae5353-1548-434a-af82-d3eacd95a0cc">
<img width="1183" alt="7" src="https://github.com/wlcrs/huawei_solar/assets/77939584/a026ab0c-bc49-43cd-9a9f-433141305c8e">
<img width="469" alt="8" src="https://github.com/wlcrs/huawei_solar/assets/77939584/5d54712e-68d6-447a-b2f1-e8af7dbe24af">
<img width="1089" alt="9" src="https://github.com/wlcrs/huawei_solar/assets/77939584/e55fd649-7978-4cd8-885a-1b6573aa0dd2">

Logger: huawei_solar.huawei_solar
Source: /usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py:184
First occurred: February 3, 2024 at 9:29:47 PM (8 occurrences)
Last logged: February 3, 2024 at 10:50:20 PM

Aborting client creation due to error.
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 178, in create
    await huawei_solar._initialize()
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 107, in _initialize
    self.time_zone = (await self.get(rn.TIME_ZONE)).value
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 241, in get
    return (await self.get_multiple([name], slave))[0]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 276, in get_multiple
    response = await self._read_registers(registers[0].register, total_length, slave)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 398, in _read_registers
    return await _do_read()
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/backoff/_async.py", line 151, in retry
    ret = await target(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/backoff/_async.py", line 151, in retry
    ret = await target(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huawei_solar/huawei_solar.py", line 351, in _do_read
    response = await self._client.read_holding_registers(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymodbus/client/base.py", line 207, in async_execute
    raise ModbusIOException(
pymodbus.exceptions.ModbusIOException: Modbus Error: [Input/Output] ERROR: No response received after 3 retries

**Comment by miguelcrego** [comment]
Try change "Flow control settings" to "Half Duplex" and let me know.
![image](https://github.com/wlcrs/huawei_solar/assets/11590052/79d3044b-0194-4fc8-bf83-099e80b4f4a5)

**Comment by szenna** [comment]
Thanks for the idea. I tried it, but unfortunately the same thing happens. Do you have any more ideas?

**Comment by marvinp93** [comment]
Have you found a solution? I have exactly the same problem with an EW11

**Reply by szenna** [reply]
Unfortunately I haven't found a solution yet...

**Reply by LA823E** [reply]
sililar Problem here. I even tried to connect without the SDongle, but no luck as well. I got some sent bytes but can not set up the connection.

**Comment by rtosa** [comment]
Do you have an active SDongle connected to your inverter? I'm asking because I have the exact same issue, and I've just seen this in the documentation:

```
The SDongleA-05 and SDongleA-FE dongles interfere with devices connected to the RS485A1 and RS485B1 pins on the inverter.
If your inverter has such a dongle installed you cannot use a serial connection option. We recommend connecting to the Modbus-TCP port exposed by that dongle instead.
```
Source: https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-the-inverter

Haven't tried without the SDongle, though.
